tests/transition.md

```diff
@@ -659,3 +662,174 @@ All test cases described in this document are beginning in a pre-Merge world and
+  
+  </details>
+
+* [ ] Transition on a valid chain with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- TB <- P1 <- P2 <- P3`, `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
+  * `B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * EL starts with `A` and `B` imported
+  * `newPayload(P1)`
+    * EL returns `{status: VALID, latestValidHash: payload.blockHash}`
+    * EL's head points to `TB`
+  * `forkchoiceUpdated(head: P1, safe: 0x00..00, finalized: 0x00..00)`
+    * EL returns `{status: VALID, latestValidHash: forkchoiceState.headBlockHash}`
+    * EL's head updated to `P1`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P2) + forkchoiceUpdated(head: P2, safe: P1, finalized: 0x00..00)`
+    * EL's head updated to `P2`
+    * `eth_getBlockByNumber(safe)` returns `-39001: Unknown block` error
+    * `eth_getBlockByNumber(finalized)` returns `-39001: Unknown block` error
+  * `newPayload(P3) + forkchoiceUpdated(head: P3, safe: P2, finalized: P1)`
+    * EL's head updated to `P3`
+    * `eth_getBlockByNumber(safe)` returns `P2`
+    * `eth_getBlockByNumber(finalized)` returns `P1`
+  
+  </details>
+
+* [ ] Transition on an invalid chain with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `Genesis <- ... <- INV_TB <- P1`, `INV_TB` is a *valid PoW* but an *invalid terminal* block:
+    * [ ] `INV_TB.TD < TTD`
+    * [ ] `INV_TB.parent.TD >= TTD` -- it might require a second EL with a higher `TTD` value
+  * EL starts with `INV_TB` as the head
+  * `newPayload(P1)`
+    * EL returns `{status: INVALID, latestValidHash: 0x00..00}`
+    * EL's head points to `INV_TB` if `INV_TB.TD < TTD`, and to `INV_TB.parent` if `INV_TB.parent.TD >= TTD`
+  
+  </details>
+
+* [ ] Syncing with the chain having a valid transition,  with enabled `TBH` override
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * `A: Genesis <- ... <- Bn <- TB <- P1`, `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
+  * `B: Genesis <- ... <- Bn`, `B` chain is heavier than `A` and would be canonical unless `TBH` override is activated
+  * EL starts with `B` imported
+  * `newPayload(P1) + forkchoiceUpdated(P1)`
+    * EL returns `{status: SYNCING}`
+    * EL's head points to `Bn`
+  * EL should pull `TB <- P1` from the network
+  * poll `forkchoiceUpdated(P1)`
+    * EL returns `{status: VALID, latestValidHash: null}`
+    * EL's head points to `P1`
+  
+  </details>
+
+* [ ] Stop processing synced PoW blocks with `TBH` override enabled
+  <details>
+  <summary>Click for details &#9662;</summary>
+
+  * PoW client starts with `Genesis <- ... <- TB <- B1 <- B2 <- ... <- Bn` chain, where `TB` is a valid terminal block, i.e. `TB.blockHash == TBH`
```

Comment:
For this case, is `TERMINAL_TOTAL_DIFFICULTY <= TB.TotalDifficulty` ? Or how is the TTD configured ?

Reply:
Yes, the following condition should hold: `TERMINAL_TOTAL_DIFFICULTY == TB.TotalDifficulty`